RFCs/polka-index.md

```diff
@@ -0,0 +1,79 @@
+# RFC-001: Polka-index
+
+## Abstract
+
+This document describes Polka-index, a indexing application that maps Content Identifiers (CIDs) to the Peer IDs of storage providers in a peer-to-peer (P2P) network. Users interact with Polka-index through a simple JSON API, while Polka-Index communicates with storage providers over a P2P network using [libp2p](https://docs.libp2p.io/). This document outlines the system's architecture, key components, and communication protocols.
+
+## Introduction
+
+In the polka-storage system it can be hard to find which CID is owned by which storage provider. Polka-index solves this problem by connecting CIDs (unique identifiers for files) to the Peer IDs of storage providers who store them.
+
+Storage providers store files outside the scope of the P2P network. Instead, they use the network to broadcast their Peer IDs and respond to CID queries from the Polka-Index. Users can easily request CID-to-Peer ID mappings from the Polka-Index.
+
+This document explains how the system works, detailing how users, the Polka-Index, and storage providers interact in the network.
+
+## Terminology
+
+- **CID (Content Identifier)**: A unique identifier for a file stored by a storage provider.
+- **Peer ID**: A unique identifier for a node in the P2P network.
+- **Storage Provider**: A node in the network that stores files and manages a database of CIDs but uses the P2P network only to share Peer IDs and provide CID mappings.
+- **Deal Database**: A local database in each storage provider where CIDs and metadata about stored files are recorded.
+- **Polka-Index**: The node responsible for querying storage providers and maintaining a database of CID-to-Peer ID mappings. It also provides an JSON API for users.
+
+## System Overview
+
+### Bootstrap Nodes
```

Comment:
Who's gonna run those bootstrap nodes as well as the previous question I asked, who's gonna run the Indexer nodes?
We need 'explanation in text' of our reasoning.

Reply:
I think storage providers have an incentive to run the bootstrap nodes because if they run a well-known bootstrap node, it is good advertising to them for storage clients, making them seem more trustworthy as they run a service that helps clients discover where content is located. When running a bootstrap node, there is an incentive to run an indexer node, because they want content to be discovered inside the P2P network.

Reply:
Seems reasonable. But open an "incentives" section that covers the indexer and bootstrap